src/metrics/nom.rs

```diff
@@ -68,6 +68,96 @@ impl Stats {
     }
 }
 
+#[inline(always)]
+fn is_child_identifier(node: &Node, id: u16) -> bool {
+    let mut cursor = node.object().walk();
+    for child in node.object().children(&mut cursor) {
+        if child.kind_id() == id {
+            return true;
+        }
+    }
+    false
+}
+
+macro_rules! function {
+    ($node: ident, $stats: ident) => {
+        // Consider named functions as functions and unnamed ones as
+        // closures.
+        if count_specific_ancestors!(
+            $node,
+            VariableDeclarator | AssignmentExpression | LabeledStatement | Pair,
+            StatementBlock | ReturnStatement | NewExpression | Arguments
+        ) > 0
+            || is_child_identifier($node, Identifier as u16)
+        {
+            $stats.functions += 1;
+        } else {
+            $stats.closures += 1;
+        }
+    };
+}
+
+macro_rules! arrow_function {
+    ($node: ident, $stats: ident) => {
+        // Consider named functions as functions and unnamed ones as
```

Comment:
Same comment as macro above, but the macros are slightly different so the comment should reflected that.